examples/TUTORIAL.md

```diff
@@ -0,0 +1,253 @@
+#
+
+if [ "$1" = "-h" ]; then
+	# If the 'nginx-centos7' assemble script is executed with '-h' flag,
+	# print the usage.
+	exec /usr/libexec/s2i/usage
+fi
+
+# Restore artifacts from the previous build (if they exist).
+#
+if [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
+  echo "---> Restoring build artifacts"
+  mv /tmp/artifacts/. ./
+fi
+
+echo "---> Installing application source"
+cp -Rf /tmp/src/. ./
+
+```
+
+There's also a usage script that lets the user know how he can run the intended image.
+
+What we need to do is tell the run script what process to run on start.
+
+```bash
+#!/bin/bash -e
+#
+# S2I run script for the 'nginx-centos7' image.
+# The run script executes the server that runs your application.
+#
+# For more information see the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+
+# We will turn off daemonizing for the nginx process so that the container
+# doesn't exit after the process runs.
+# We will also specify our configuration that we copied in the builder image.
+exec /usr/sbin/nginx -c /opt/app-root/etc/nginx.conf -g "daemon off;"
+```
+Next, we continue by creating our base builder image with make.
+
+In the current project directory (`source-to-image/examples/s2i-nginx`) just issue the `make` command. Make creates our base builder image from the instructions in our `Dockerfile`. After it finishes we can use our builder image to test out the webserver configuration.
+
+```
+[user@source-to-image s2i-nginx] $ make
+docker build -t nginx-centos7 .
+Sending build context to Docker daemon 27.65 kB
+Step 0 : FROM openshift/base-centos7
+ ---> f8b3a23256c3
+Step 1 : MAINTAINER Victor Palade <ipalade@redhat.com>
+ ---> Using cache
+ ---> 2d23d0e4e9d8
+Step 2 : ENV NGINX_VERSION 1.6.3
+ ---> Using cache
+ ---> 7d3fcf7a2df8
+Step 3 : LABEL io.k8s.description "Nginx Webserver" io.k8s.display-name "Nginx 1.6.3" io.openshift.expose-services "8080:http" io.openshift.tags "builder,webserver,html,nginx"
+ ---> Using cache
+ ---> 73e7f639c873
+Step 4 : RUN yum install -y nginx && yum clean all
+ ---> Using cache
+ ---> c18ab3359cd6
+Step 5 : COPY ./.s2i/bin/ /usr/libexec/s2i
+ ---> 30d5cd79c482
+Removing intermediate container 23af1710796a
+Step 6 : COPY ./etc/ /opt/app-root/etc
+ ---> 4caa2fa485d0
+Removing intermediate container 79fe46953e2f
+Step 7 : RUN chown -R 1001:1001 /var/log/nginx/ && chown -R 1001:1001 /var/lib/nginx/     && mkdir -p /opt/app-root/run && chown -R 1001:1001 /opt/app-root/run
+ ---> Running in ce4c95420fbf
+ ---> e6e1ab3cc77c
+Removing intermediate container ce4c95420fbf
+Step 8 : USER 1001
+ ---> Running in 8422927dfd36
+ ---> ceb6f4c86b5b
+Removing intermediate container 8422927dfd36
+Step 9 : EXPOSE 8080
+ ---> Running in f10b4055dbcd
+ ---> 2b234ffa6558
+Removing intermediate container f10b4055dbcd
+Step 10 : CMD usage
+ ---> Running in dd8a407c9441
+ ---> f81e42101d9c
+Removing intermediate container dd8a407c9441
```

Comment:
Shorten the listing as much as possible, it's not necessary to have it here. You can do:
```
[user@source-to-image s2i-nginx] $ make
docker build -t nginx-centos7 .
...
Successfully built f81e42101d9c
```